### PR TITLE
[CC-31] Create branch when transifex updated

### PR DESCRIPTION
**Pull request by dpoirier**
## Description
Create git branch when transifex updated, add changed translation files, commit, and push upstream.

## Checklist
- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x] My pull request targets the *default* branch of the repository (`main` or `master`).
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: licenses/git_utils.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        print("local already exists")
+    branch = getattr(repo.heads, branch_name)
+    branch.checkout()
+    # Make sure local branch is up to date if there's an upstream branch
+    if hasattr(origin.refs, branch_name):
+        origin.pull(f"{branch_name}:{branch_name}")
+
+
+def commit_and_push_changes(repo: git.Repo, branch_name: str, commit_msg: str):
+    """Commit new translation changes to current branch, and push upstream"""
```
Comment (Audiosutras) [inline licenses/git_utils.py:33]: It looks like we can commit to any branch through this definition. Can we remove `"current"` from the docstring.

File: licenses/transifex.py [hunk 2]
```diff
@@ -164,3 +164,125 @@ def upload_messages_to_transifex(self, legalcode, pofile: polib.POFile = None):
...
+            resource_url = f"{fetch_resources_url}{resource_slug}"
+            r = self.request25("get", resource_url)
+            stats[resource_slug] = r.json()["stats"]
+        return stats
+
+    def check_for_translation_updates(self):
+        """
+        Use the Transifex API to find the last update timestamp for all our translations.
+        If translations are updated, we might end up creating a branch or other actions.
+        """
```
Comment (Audiosutras) [inline licenses/transifex.py:186]: I think we can update this docstring?

File: licenses/git_utils.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    index = repo.index
+    index.commit(commit_msg)
+    results = repo.remotes.origin.push(f"{branch_name}:{branch_name}")
```
Comment (Audiosutras) [inline licenses/git_utils.py:36]: In this function should we use `call_command` and run `publish` for the git branch that has been updated?

File: licenses/git_utils.py (hunk 1 above)
Comment (dpoirier) [reply licenses/git_utils.py:33]: Good point - it's actually committing the changes to the current branch, then pushing whatever branch was passed in, which doesn't make a lot of sense. I'll change it to just work on the current branch without passing in a name.

File: licenses/git_utils.py (hunk 3 above)
Comment (dpoirier) [reply licenses/git_utils.py:36]: Indeed, we should!

File: licenses/transifex.py (hunk 2 above)
Comment (dpoirier) [reply licenses/transifex.py:186]: Will expand.

File: licenses/git_utils.py (hunk 3 above)
Comment (dpoirier) [reply licenses/git_utils.py:36]: Actually I'm thinking we call publish from the management command, and just have this function do git stuff.

Comment (Audiosutras) [reply licenses/git_utils.py:38]: copy that

File: licenses/management/commands/check_for_translation_updates.py [hunk 4]
```diff
@@ -1,8 +1,20 @@
...
 class Command(BaseCommand):
     def handle(self, **options):
-        check_for_translation_updates()
+        branches_updated = TransifexHelper().check_for_translation_updates()
+
+        # run collectstatic if we're going to publish
+        if branches_updated:
+            call_command("collectstatic", interactive=False)
+
+        for branch_name in branches_updated:
```
Comment (Audiosutras) [inline licenses/management/commands/check_for_translation_updates.py:18]: ✔️

File: licenses/management/commands/check_for_translation_updates.py [hunk 5]
```diff
@@ -1,8 +1,20 @@
...
+
+TOP_BRANCH = "develop"
+
+branch_name = "test_branch"
 
 
 class Command(BaseCommand):
     def handle(self, **options):
-        check_for_translation_updates()
+        branches_updated = TransifexHelper().check_for_translation_updates()
```
Comment (Audiosutras) [inline licenses/management/commands/check_for_translation_updates.py:12]: I think we should log to the cli how branches are to be updated

File: licenses/management/commands/check_for_translation_updates.py [hunk 6]
```diff
@@ -1,8 +1,20 @@
...
+
+        # run collectstatic if we're going to publish
```
Comment (Audiosutras) [inline licenses/management/commands/check_for_translation_updates.py:14]: Should we log when we are running collectstatic and that we are publishing updated branches?

**Review by Audiosutras — APPROVED** [review]
I left a couple comments, but nothing that stops approval and merging. 👍

**Pull request merged**